### PR TITLE
Scoped changes place painting on ar surface

### PR DESCRIPTION
**Pull request by rushitvaghela-vc**
Changed AR view Feature as Client Requirement scoped
solved back button issue native side

### CODE REVIEW

File: Script/Native/UnityReceiver.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            AndroidJavaObject currentActivity = unityPlayer.GetStatic<AndroidJavaObject>("currentActivity");
+
+            AndroidJavaObject context = currentActivity.Call<AndroidJavaObject>("getApplicationContext");
+
+            MyBroadcastReceiver receiver = new MyBroadcastReceiver();
+            receiver.Register(context);
+        }
+
+        public class MyBroadcastReceiver : AndroidJavaProxy {
+            private static readonly string ACTION = "com.example.MY_UNITY_ACTION";
```
Comment (vinayak-vc) [inline Script/Native/UnityReceiver.cs:17]: this line has "example" does it need to change?

File: Script/TestBounds.cs
Comment (vinayak-vc) [inline Script/TestBounds.cs:1]: remove this file if not needed

File: Script/Manager/ThreeDARFrameCanvas.cs [hunk 2]
```diff
@@ -76,6 +81,33 @@ public class ThreeDARFrameCanvas : MonoBehaviour {
...
+
+
+
+            ResizeOutofBoundImage();
+            //StartCoroutine(ResizeOutofBoundImage());
+        }
+
+
+        private void ResizeOutofBoundImage() {
+            //screenBounds = Camera.main.ScreenToWorldPoint(new Vector3(Screen.width, Screen.height, Camera.main.transform.position.z));
```
Comment (vinayak-vc) [inline Script/Manager/ThreeDARFrameCanvas.cs:95]: remove this comments

File: Script/Manager/ThreeDARFrameCanvas.cs [hunk 3]
```diff
@@ -76,6 +81,33 @@ public class ThreeDARFrameCanvas : MonoBehaviour {
 
             RectTransform rectTransform = bgFrame.GetComponent<RectTransform>();
             rectTransform.sizeDelta = new Vector2(newWidth, newHeight);
+
+
+
+
+
+            ResizeOutofBoundImage();
+            //StartCoroutine(ResizeOutofBoundImage());
```
Comment (vinayak-vc) [inline Script/Manager/ThreeDARFrameCanvas.cs:90]: remove this comments

File: Script/Manager/LobbyManager.cs [hunk 4]
```diff
@@ -190,9 +219,10 @@ public enum URL_type {
             //StartCoroutine(DownloadImageURL(imageURL));
```
Comment (vinayak-vc) [inline Script/Manager/LobbyManager.cs:219]: remove this comments

File: Script/Manager/LobbyManager.cs [hunk 5]
```diff
@@ -156,7 +180,9 @@ public enum URL_type {
             if (no == 0) {
                 SceneManager.LoadScene("Main-AR");
             } else if (no == 1) {
-                SceneManager.LoadScene("Main-ARFrame");
+                //SceneManager.LoadScene("Main-ARFrame Backup ARCore");
```
Comment (vinayak-vc) [inline Script/Manager/LobbyManager.cs:183]: remove this comments

File: Script/Manager/LobbyManager.cs [hunk 6]
```diff
@@ -52,10 +53,10 @@ public enum URL_type {
                     // return "https://gallerieapi.imgix.net/Product/3-61.webp"; //Divya
                     //return "https://gallerieapi.imgix.net/Product/3-61.jpg"; //Divya //32BitDepth Converted Not Working
                     //return "https://3d-model-construction.s3.ap-south-1.amazonaws.com/frame_0000.png"; //Parth S3 Bucket
-                    //return "https://drive.google.com/uc?export=download&id=1f82_wC78r58w3GGTT8fjyxe2Vn2gT6Nc"; //Divya
+                    return "https://drive.google.com/uc?export=download&id=1f82_wC78r58w3GGTT8fjyxe2Vn2gT6Nc"; //Divya
                     // return "https://drive.google.com/uc?export=download&id=1lgHhQLV6vl92p8x62teyqHHGsfGtXObI"; //Divya
```
Comment (vinayak-vc) [inline Script/Manager/LobbyManager.cs:57]: remove this comments

File: Script/Manager/LobbyManager.cs [hunk 7]
```diff
@@ -52,10 +53,10 @@ public enum URL_type {
                     // return "https://gallerieapi.imgix.net/Product/3-61.webp"; //Divya
```
Comment (vinayak-vc) [inline Script/Manager/LobbyManager.cs:53]: remove this comments

File: Script/AR/ARFrame/ARPlaceObjectOnSurface.cs [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+namespace ViitorCloud.ARModelViewer {
+    public class ARPlaceObjectOnSurface : MonoBehaviour {
+        public ARRaycastManager m_RaycastManager;
+        private List<ARRaycastHit> hits = new List<ARRaycastHit>();
+        public GameObject objectToPlace;
+
+
+        // Start is called before the first frame update
+        void Start() {
```
Comment (vinayak-vc) [inline Script/AR/ARFrame/ARPlaceObjectOnSurface.cs:16]: remove this empty method

File: Script/AR/ARFrame/ARFramePlacementManager.cs [hunk 9]
```diff
@@ -0,0 +1,338 @@
...
+        }
+
+        #endregion SwipeLogic
+
+        #region Test Dimension Panel
+
+        [SerializeField] private TMPro.TMP_InputField _inputField_Width;
+        [SerializeField] private TMPro.TMP_InputField _inputField_Height;
+        [SerializeField] private TMPro.TMP_InputField _inputField_Unit;
+        //[SerializeField] private TMPro.TMP_InputField _inputField_Speed;
```
Comment (vinayak-vc) [inline Script/AR/ARFrame/ARFramePlacementManager.cs:323]: remove this comment

File: Script/AR/ARFrame/ARFramePlacementManager.cs [hunk 10]
```diff
@@ -0,0 +1,338 @@
...
+            //IntPtr jmethod = AndroidJNI.GetMethodID(jclass, "finish", "()V");
+            //AndroidJNI.CallVoidMethod(jactivity, jmethod, new jvalue[] { });
+
+            //// Release references to avoid memory leaks
+            //AndroidJNI.DeleteLocalRef(jactivity);
+            //AndroidJNI.DeleteLocalRef(jclass);
+        }
+        private void CallBackFromNative() {
+            //AndroidJavaClass unityPlayer = new AndroidJavaClass("com.unity3d.player.UnityPlayer");
+            //AndroidJavaObject currentActivity = unityPlayer.GetStatic<AndroidJavaObject>("currentActivity");
```
Comment (vinayak-vc) [inline Script/AR/ARFrame/ARFramePlacementManager.cs:262]: remove this comments

File: Script/AR/ARFrame/ARFramePlacementManager.cs [hunk 11]
```diff
@@ -0,0 +1,338 @@
...
+        public IEnumerator CloseActivity() {
+            Debug.Log("-- close activity start--");
+            arPlaneManager.gameObject.SetActive(false);
+            yield return new WaitForEndOfFrame();
+
+            CallBackFromNative();
+            Debug.Log("-- close activity finished--");
+        }
+
+        private void FinishActivityNativeOLD() {
```
Comment (vinayak-vc) [inline Script/AR/ARFrame/ARFramePlacementManager.cs:247]: remove this empty method

File: Script/AR/ARFrame/ARFramePlacementManager.cs [hunk 12]
```diff
@@ -0,0 +1,338 @@
...
+                //DestroyImmediate(spawnedObject);
+                //spawnedObject = null;
+                //RemoveAllDetectedPlanes();
+                ////Reopen UI
+                //SetupInitial();
+                SceneManager.LoadScene(SceneManager.GetActiveScene().name);
+            }
+        }
+
+        //private void RemoveAllDetectedPlanes() {
```
Comment (vinayak-vc) [inline Script/AR/ARFrame/ARFramePlacementManager.cs:216]: remove this comments

File: Script/AR/ARFrame/ARFramePlacementManager.cs [hunk 13]
```diff
@@ -0,0 +1,338 @@
...
+        }
+
+        public void OnButtonReset() {
+            //ResetPositionRottion();
+            RemoveFrame();
+        }
+
+        private void RemoveFrame() {
+            if (spawnedObject != null) {
+                //DestroyImmediate(spawnedObject);
```
Comment (vinayak-vc) [inline Script/AR/ARFrame/ARFramePlacementManager.cs:207]: remove this comments

File: Script/AR/ARFrame/ARFramePlacementManager.cs [hunk 14]
```diff
@@ -0,0 +1,338 @@
...
+        }
+
+        /// <summary>
+        /// The object instantiated as a result of a successful raycast intersection with a plane.
+        /// </summary>
+        public GameObject spawnedObject {
+            get; private set;
+        }
+
+        private void Awake() {
```
Comment (vinayak-vc) [inline Script/AR/ARFrame/ARFramePlacementManager.cs:69]: remove this empty method

File: Script/AR/ARFrame/ARFrame3D.cs [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace ViitorCloud.ARModelViewer {
+    public class ARFrame3D : MonoBehaviour {
+        [SerializeField] private Material paintingMaterial;
+        [SerializeField] private Material frameMaterial;
+        [SerializeField] private Transform frame_transform;
+
+        //[SerializeField] private Image frameImageWhite;
```
Comment (vinayak-vc) [inline Script/AR/ARFrame/ARFrame3D.cs:12]: remove this comment

**Review by vinayak-vc — CHANGES_REQUESTED** [review]

File: Script/Native/UnityReceiver.cs (hunk 1 above)
Comment (rushitvaghela-vc) [reply Script/Native/UnityReceiver.cs:17]: removed this script from scene for now 
if not needed delete this script

**Pull request merged**